src/components/UpdatePassword/UpdatePasswordButton.tsx

```diff
@@ -0,0 +1,136 @@
+import { useContext, useRef, useState } from "react";
+import { updatepasswordService } from "../../services/auth/updatepassword.service";
+import { FieldValues, useForm } from "react-hook-form";
+import { AuthContext } from "../../context/AuthContext";
+import toast from "react-hot-toast";
+
+export function UpdatePassword() {
+  const { session } = useContext(AuthContext);
+  const [showModal, setShowModal] = useState(false);
+  const formRef = useRef<HTMLFormElement>(null);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm();
+
+  interface UpdatepasswordRequest {
+    oldPassword: string;
+    newPassword: string;
+    token: string;
+  }
+
+  const onSubmit = async (formData: FieldValues, token: string) => {
+    const req: UpdatepasswordRequest = {
+      oldPassword: formData.password,
+      newPassword: formData.newPassword,
+      token: token
+    };
+    if (!errors.password && !errors.newPassword) {
+      try {
```

Comment:
The `try-catch` is not needed. Since you're using a `try-catch` in the service any error will be caught there.

Reply:
Done